### PR TITLE
Order tests

### PR DESCRIPTION
**Pull request by joshuataylor**
This takes PR https://github.com/commerceguys/commerce/pull/142 , and adds order tests, order admin tests and fixes the YML files.

### CODE REVIEW

File: modules/order/src/Tests/CommerceOrderTestBase.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace Drupal\commerce_order\Tests;
+
+use Drupal\commerce_product\Entity\Product;
+use Drupal\commerce_store\Entity\Store;
+use Drupal\commerce_store\Tests\StoreTestBase;
+use Drupal\Component\Utility\SafeMarkup;
+use Drupal\simpletest\WebTestBase;
+
+/**
+ * Defines base class for shortcut test cases.
```
Comment (bojanz) [inline modules/order/src/Tests/CommerceOrderTestBase.php]: Not shortcut :)

File: modules/order/src/Tests/CommerceOrderTestBase.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ */
+abstract class CommerceOrderTestBase extends WebTestBase {
+
+  public $product;
```
Comment (bojanz) [inline modules/order/src/Tests/CommerceOrderTestBase.php]: These two variables can be protected, and need documentation.

File: modules/order/src/Tests/CommerceOrderTestBase.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  public static $modules = [
+    'commerce',
+    'commerce_order',
+    'commerce_price',
+    'commerce_line_item',
+    'inline_entity_form'
+  ];
+
+  /**
+   * User with permission to administer products.
```
Comment (bojanz) [inline modules/order/src/Tests/CommerceOrderTestBase.php]: Missing @var

File: modules/order/src/Tests/CommerceOrderTestBase.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      'access administration pages',
+    ));
+    // Create a store
+    $values = [
+      'name' => t('Default store'),
+      'uid' => 1,
+      'mail' => \Drupal::config('system.site')->get('mail'),
+      'type' => 'default',
+      'default_currency' => "USD",
+      'currency' => "USD",
```
Comment (bojanz) [inline modules/order/src/Tests/CommerceOrderTestBase.php]: there's no "Currency" key anymore. Also, don't we need to set the address? Maybe not

**Comment by joshuataylor** [comment]
Updated PR as per feedback

**Pull request merged**